Makefile

```diff
@@ -7,19 +7,23 @@ APP_NAME := raven-wps
 
 OS := $(shell uname)
 
-WPS_PORT := 9099
-WPS_URL := http://0.0.0.0:$(WPS_PORT)
+WPS_PORT ?= 9099
+WPS_URL ?= http://0.0.0.0:$(WPS_PORT)
```

Comment:
Makefile noob question, what's the difference between `:=` and `?=`?  The former `:=` already works fine by the way.

Reply:
`:=` sets the value, `?=` sets the value if not already set (good for defaults)

There's also a `=` that does the same as `:=`, but allows dynamic resolution of the right part only when referenced somewhere else by `$( )` instead of on the line the `=` is indicated. This allows, for example, computing a `$(WPS_PORT)` value later on, and still obtain the desired result of the "final" value.